security/pkg/pki/util/keycertbundle_test.go

```diff
@@ -484,3 +485,97 @@ func TestExtractCACertExpiryTimestamp(t *testing.T) {
 		})
 	}
 }
+
+func TestTimeBeforeNextRotate(t *testing.T) {
+	t0 := time.Now()
+	caCertTTL := time.Second * 60
+	rootCertBytes, rootKeyBytes, err := GenCertKeyFromOptions(CertOptions{
+		Host:         "citadel.testing.istio.io",
+		Org:          "MyOrg",
+		NotBefore:    t0,
+		IsCA:         true,
+		IsSelfSigned: true,
+		TTL:          time.Hour,
+		RSAKeySize:   2048,
+	})
+	if err != nil {
+		t.Errorf("failed to gen root cert for Citadel self signed cert %v", err)
+	}
+
+	rootCert, err := ParsePemEncodedCertificate(rootCertBytes)
+	if err != nil {
+		t.Errorf("failed to parsing pem for root cert %v", err)
+	}
+
+	rootKey, err := ParsePemEncodedKey(rootKeyBytes)
+	if err != nil {
+		t.Errorf("failed to parsing pem for root key cert %v", err)
+	}
+
+	caCertBytes, _, err := GenCertKeyFromOptions(CertOptions{
+		Host:         "citadel.testing.istio.io",
+		Org:          "MyOrg",
+		NotBefore:    t0,
+		TTL:          time.Second * 60,
+		IsServer:     true,
+		IsCA:         true,
+		IsSelfSigned: false,
+		RSAKeySize:   2048,
+		SignerCert:   rootCert,
+		SignerPriv:   rootKey,
+	})
+	if err != nil {
+		t.Fatalf("failed to gen CA cert for Citadel self signed cert %v", err)
+	}
+
+	testCases := []struct {
+		name         string
+		cert         []byte
+		workloadTTL  time.Duration
+		expectedTime time.Duration
+		expectedErr  error
+	}{
+		{
+			name:         "workload TTL higher than cert TTL, so it should return cert TTL",
```

Comment:
To be accurate, can you revise "cert TTL" (there are multiple of them in this test) to be "CA cert TTL"?